src/trim.jl

```diff
@@ -0,0 +1,55 @@
+"""
+    getinternalnodes(t::AbstractTree)
+Function to retrieve only the internal nodes from a tree, `t`, which does not
+include tips or root.
+
+"""
+function getinternalnodes(t::AbstractTree)
+    return collect(nodenamefilter(x->!isleaf(x)& !isroot(x), t))
+end
+"""
+    drop_tip!(t::AbstractTree, tip::Vector{String})
+Function to drop tips from a phylogenetic tree `t`, which are not found in
+the vector of tip names, `tip`.
+
+"""
+function drop_tip!(t::AbstractTree, tip::Vector{String})
```

Comment:
You need to export drop_tip!() in the `Phylo.jl` file. Also I would go for `tips` as the argument rather than `tip` since it can refer to multiple tips.

Reply:
In fact now I think about it, `droptips!()` would be better too - this package tends not to have underscores in function names, and you may be dropping multiple tips again...

Reply:
Also, don't forget that the label type of the tips can vary, so `AbstractTree{NodeLabel, BranchLabel}` tells you what the `NodeLabel` type is...